crates/proto/src/op/message.rs

```diff
@@ -665,12 +667,30 @@ impl Message {
                 continue;
             }
 
-            match record.record_type() {
+            match record.data() {
                 #[cfg(feature = "__dnssec")]
-                RecordType::SIG => sig = MessageSignature::Sig0(record),
+                RData::DNSSEC(DNSSECRData::SIG(_)) => {
+                    sig = MessageSignature::Sig0(
+                        record
+                            .map(|data| match data {
+                                RData::DNSSEC(DNSSECRData::SIG(sig)) => Some(sig),
+                                _ => None,
+                            })
+                            .unwrap(),
```

Comment:
Can you add a comment justifying the `unwrap()` here and on L690? AFAICT both are infallible because of the `match record.data()` ?